File: app/src/main/java/com/temperoni/recipes/mvp/view/RecipesListView.java
package com.temperoni.recipes.mvp.view;

import com.temperoni.recipes.ui.models.RecipeViewModel;

import java.util.List;

/**
 * @author Leandro Temperoni
 */

public interface RecipesListView {

    void displayRecipes(List<RecipeViewModel> recipes);

    void navigateToRecipeDetail(int recipeId);
}

File: app/src/main/java/com/temperoni/recipes/mvp/view/RecipesListView.kt
package com.temperoni.recipes.mvp.view

import android.view.View

import com.temperoni.recipes.ui.models.RecipeViewModel

/**
 * @author Leandro Temperoni
 */

interface RecipesListView {

    fun displayLoading()

    fun displayError()

    fun displayEmptyState()

    fun displayRecipes(data: List<RecipeViewModel>)

    fun navigateToRecipeDetail(recipeId: String, imageUrl: String, sharedView: View)
}
